### ISSUE TITLE
React native CSM support

### ISSUE THREAD

**Issue by WRiferreiro** [amazon-connect/amazon-connect-chatjs#171]
Hi everyone,

While support for react native was added as part of #153, some features of ChatJS like client side metrics (csm) are not yet supported.

**When the SDK initializes the csm module to following log is shown:**
`ChatJS-csmService: Failed to addLatencyMetric csm:  ReferenceError: Property 'csm' doesn't exist undefined`

Which seems to be originated from here: https://github.com/amazon-connect/amazon-connect-chatjs/blob/master/src/service/csmService.js#L32

**Subsequent calls to the csm module to add metrics also result in an error:**
`ChatJS-csmService: Failed to addCountAndErrorMetric csm:  ReferenceError: Property 'csm' doesn't exist undefined`

Is there a plan to add csm support for react native?

**Comment by spencerlepine** [comment]
Hi @WRiferreiro, thank you for taking the time to raise this issue.

Conducting further investigation into this to see if we can easily fix. Please note team does not have ETA to resolve this, and CSM support in React Native will not be supported yet.

Confirming this is real bug - `connect-csm.js` has not been updated to support ReactNative. CSM will not work in `amazon-connect-chatjs@^1.5.0` because it is trying to add a script tag to the DOM, which is not possible in React Native.

Does this answer your question?

**Comment by spencerlepine** [comment]
For now, to avoid logging please disable CSM:

```js
const customerChatSession = connect.ChatSession.create({
  ...,
  disableCSM: true
});
```

---

To avoid further confusion, I will update documentation. 

If there are any updates from team, we will provide comments on this issue.
 
 Thanks,
 Spencer

**Comment by spencerlepine** [comment]
Hi @WRiferreiro, 

There was a bug causing `ERROR` logs even when CSM was disabled (also reproducible in browser environment). I have raise bug fix in #175 to address this.

**Regarding CSM in React Native:**
Team will discuss this and add task to our backlog if we decided to support ChatJS CSM in React Native. There is not official support for the `connect-csm.js` in browser environment, so there would need to be effort to migrate + maintain new file.

Please feel free to reach out if you have further questions.

Thanks,
Spencer

**Comment by spencerlepine** [comment]
Hi WRiferreiro, thank you for taking the time to raise this issue. You have uncovered an existing bug that will be addressed (in PR https://github.com/amazon-connect/amazon-connect-chatjs/pull/175 as mentioned above), which is not directly related to React Native.

I have reviewed this with team, and there is no further action item we are taking for this issue. Client-side-metric code is for product internal use only, and we are not supporting an CSM features for React Native currently.

Going to resolve this issue to avoid confusion. This behavior will not impact ChatJS functionality and you can safely disable CSM in the `setGlobalConfig()`

Thanks,
Spencer
